tests/tracer/test_memory_leak.py

```diff
@@ -0,0 +1,80 @@
+"""
+Variety of test cases ensuring that ddtrace does not leak memory.
+"""
+import gc
+import os
+from typing import TYPE_CHECKING
+from weakref import WeakValueDictionary
+
+import pytest
+
+from ddtrace import Tracer
+
+
+if TYPE_CHECKING:
+    from ddtrace.span import Span
+
+
+@pytest.fixture
+def tracer():
+    # type: (...) -> Tracer
+    return Tracer()
+
+
+def trace(weakdict, tracer, *args, **kwargs):
+    # type: (WeakValueDictionary, Tracer, ...) -> Span
+    """Return a reference to a span created from ``tracer.trace(*args, **kwargs)``
+    and adds it to the given weak dictionary.
+
+    Note: ensure to delete the returned reference from this function to ensure
+    no additional references are kept to the span.
+    """  # noqa: D402
```

Comment:
what is D402, not sure why this is needed?

Reply:
Yeah I wasn't sure of where to document it. For some reason it complains because of the `tracer.trace(...)`. It seems to be a bug in pydocstyle -- it thinks that the `trace` refers to this function.

https://github.com/PyCQA/pydocstyle/issues/284